proxy/reverse.go

```diff
@@ -118,6 +121,9 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 		redirectRequest(proxyreq, ep.URL)
 
 		res, err = p.transport.RoundTrip(proxyreq)
+		if requestCanceled {
```

Comment:
the two go routines might access the shared variable `requestCanceled` at the same time. It is not safe in theory.